### PR TITLE
sql, storage: misc spelling fixes

### PR DESCRIPTION
**Pull request by BramGruneir**
follow up from #8980

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8982)

### CODE REVIEW

**Comment by BramGruneir** [comment]
I'll merge when green.

File: storage/engine/engine.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type Iterator interface {
 	// is <= the provided key.
 	SeekReverse(key MVCCKey)
 	// Valid returns true if the iterator is currently valid. An
-	// iterator which hasn't been seeked or has gone past the end of the
+	// iterator which hasn't been sought or has gone past the end of the
```
Comment (tamird) [inline storage/engine/engine.go]: this is incorrect.

Comment (BramGruneir) [inline storage/engine/engine.go]: seeked is not a real word, so what would you suggest.

Comment (tamird) [inline storage/engine/engine.go]: seek is a method name, so whether or not this is a real word doesn't matter. How about `seek()`ed?

Comment (BramGruneir) [inline storage/engine/engine.go]: I was pretty sure the correct thing was to stick to normal english rules for this type of thing.

`seek()ed` isn't any better.  

Instead I'll make the comment not verbify it.
`An iterator that hasn't had Seek called on it or has gone past the end of the key range is invalid.`

**Comment by tamird** [comment]
LGTM

**Pull request merged**